### PR TITLE
Refactor tests + docker-compose file into common setup for all components

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version: '2'
+services:
+  django:
+    image: $IMAGE
+    build:
+      context: $BUILD_CONTEXT
+    ports:
+      - '8000'
+    environment:
+      - ${DATABASE_ENV}=postgres://user:password@postgres/db
```
Comment (bearnard) [inline docker-compose.yml:10]: intentional `${}` ?

Comment (bearnard) [reply docker-compose.yml:10]: Ah I see it's an env var used to create docker env var.

Comment (JayH5) [reply docker-compose.yml:10]: The `{}` are not required and a leftover from old code..

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  esac
+done
+
+# Export all the variables needed by the docker-compose file
+COMPONENT="${IMAGE##*/}"                       # praekeltfoundation/seed-message-sender -> seed-message-sender
+CLASS_NAME="$(echo "$COMPONENT" | tr '-' '_')" # seed-message-sender -> seed_message_sender
+SHORT_NAME="${PACKAGE#seed-}"                  # seed-message-sender -> message-sender
+SHORT_CLASS_NAME="${CLASS_NAME#seed_}"         # seed_message_sender -> message_sender
+
+export IMAGE="$IMAGE"
```
Comment (bearnard) [inline test.sh]: isn't `export IMAGE` sufficient?

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+wait_for_gunicorn_start django
+
+# Check that the Django admin page is accessible
+curl -sfL "$(get_service_local_address django)"/admin | fgrep '<title>Log in | Django site admin</title>'
+
+# Check the database tables were created
+[[ $(psql_cmd -q --dbname db -c \
+  "SELECT COUNT(*) FROM information_schema.tables WHERE table_schema='public';" \
+    | grep -E '^\s*[[:digit:]]+' | tr -d ' ') > 0 ]]
```
Comment (bearnard) [inline test.sh]: could get rid of the grep here by using `psql_cmd -t`?

**Comment by bearnard** [comment]
👍

File: test.sh (hunk 3 above)
Comment (bearnard) [reply test.sh]: `-P border=0` might remove the need for `tr` usage here too.

Comment (bearnard) [reply test.sh]: actually `-t -A` seems to work best.

```
echo ---$(psql -q -t -A --dbname grafana -c "SELECT COUNT(*) FROM information_schema.tables WHERE table_schema='public';")---
---20---
```

Comment (bearnard) [reply test.sh]: Since this isn't strictly new code being introduced, feel free to ignore.

File: test.sh (hunk 2 above)
Comment (JayH5) [reply test.sh]: Yes.. I always forget this, thanks.

File: test.sh (hunk 3 above)
Comment (JayH5) [reply test.sh]: Thanks.. will try

**Pull request merged**